docs/design/2021-09-29-secure-bootstrap.md

```diff
@@ -0,0 +1,82 @@
+# TiDB Design Documents
+
+- Author(s): [morgo](http://github.com/morgo)
+- Discussion PR: https://github.com/pingcap/tidb/pull/28482
+- Tracking Issue: https://github.com/pingcap/tidb/issues/28481
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Test Design](#test-design)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+Currently, TiDB will create a root user with no password and bind to `0.0.0.0/::`. This is a potential security issue if users do not have a firewall configured correctly, or if a rogue actor has local access.
+
+This document introduces the `--intialize-secure` and `--initialize-insecure` bootstrap options. The current behavior is equivalent to `--initialize-insecure`, but the intention is to change the default to secure **once** it is determined stable.
+
+## Motivation or Background
+
+The motivation for this change is to improve the default security of TiDB. The current default is insecure, and requires additional steps to be taken by users to secure TiDB.
+
+The design attempts to make the change with the least breakage possible. However, there will be some impact to usability.
+
+## Detailed Design
+
+The current bootstrap (to be referred to as `--initialize-insecure`) creates a root user as follows. This will remain unchanged:
+
+```sql
+CREATE USER 'root'@'%' IDENTIFIED WITH 'mysql_native_password' AS '' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK;
+GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' WITH GRANT OPTION;
+```
+
+The `--initialize-secure` bootstrap will instead create a user based on the OS user. For example, with an OS user of `ubuntu`:
+
+```sql
+CREATE USER 'ubuntu'@'localhost' IDENTIFIED WITH 'auth_socket' REQUIRE NONE PASSWORD EXPIRE DEFAULT ACCOUNT UNLOCK;
```

Comment:
Why use an OS dependent user instead of `root`?

We could also do something like this: 
```
CREATE USER 'root'@'localhost' IDENTIFIEW WITH 'auth_socket' AS 'ubuntu'
```
This would allow the `root` and `ubunutu` users to connect.

Reply:
We could do that too. What we would need to discuss is which is the less of 2 evils:
- Having 2 "roots"
- Having a default user being different for everyone, then instructions to create a root account.

I'm not sure tbh, they both have downsides.

Reply:
I've moved this to unresolved questions. We don't technically need to decide now.